test/configuration.test.js

```diff
@@ -72,12 +82,154 @@ QUnit.test('GOAL_BUFFER_LENGTH set warning and invalid', function(assert) {
 
   assert.equal(Config.GOAL_BUFFER_LENGTH, 30, 'default');
 
-  Hls.GOAL_BUFFER_LENGTH = 0;
+  Hls.GOAL_BUFFER_LENGTH = -1;
   assert.equal(this.env.log.warn.calls, 2, 'logged two warnings');
 
   assert.equal(Config.GOAL_BUFFER_LENGTH, 30, 'default');
 });
 
+QUnit.test('MAX_GOAL_BUFFER_LENGTH get warning', function(assert) {
```

Comment:
Although I'm usually for non shared tests, these are similar enough that it might be worth abstracting the test logic, and also checking the warning messages.